facturacion_mexico/facturacion_fiscal/tests/test_cancelacion_integridad.py

```
@@ -0,0 +1,628 @@
+	def test_13_cancelar_si_con_ffm_real_cancelado(self):
+		from facturacion_mexico.api.fiscal_operations import cancelar_si_post_fiscal
+
+		ffm = self.ffm(None, FiscalStates.CANCELADO, motivo="02")
+		# Snapshot de la SI DESACTUALIZADO (PENDIENTE) aunque la FFM real está CANCELADO.
+		si = self.si(fiscal_status=FiscalStates.PENDIENTE_CANCELACION, ffm=ffm)
+		frappe.db.set_value("Factura Fiscal Mexico", ffm, "sales_invoice", si)
+		with patch("frappe.has_permission", return_value=True), patch("frappe.get_doc") as gd:
+			# get_doc("Sales Invoice") -> obj con docstatus 1; el cancel() real se mockea.
+			real_get_doc = frappe.get_doc
+
+			def _gd(*a, **k):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_gd`

(ANN202)